Doc/library/typing.rst

```diff
@@ -1172,25 +1173,58 @@ These are not used in annotations. They are building blocks for creating generic
            """Return a list containing n references to x."""
            return [x]*n
 
-       def longest(x: A, y: A) -> A:
-           """Return the longest of two strings."""
-           return x if len(x) >= len(y) else y
 
-    The latter example's signature is essentially the overloading
-    of ``(str, str) -> str`` and ``(bytes, bytes) -> bytes``.  Also note
-    that if the arguments are instances of some subclass of :class:`str`,
-    the return type is still plain :class:`str`.
+       def print_capitalized(x: S) -> S:
+           """Print x capitalized, and return x"""
+           print(x.capitalize())
+           return x
```

Comment:
I'm not wildly keen on this example, but I was struggling to think of a better one that was (1) extremely simple and (2) showed the need for a bound `TypeVar` rather than a constrained `TypeVar`.